students/hyska_monika/lesson_06_dicts_tuples_sets_args_kwargs/functions_lesson06.py

```diff
@@ -44,30 +43,36 @@ def list_elements_str():
 def create_dict_to_n(n):
     asked_numbers = {}
     for i in range(n):
-        asked_numbers.setdefault(i + 1, 1)
+        asked_numbers[i + 1] = 1
     return asked_numbers
 
 
-# function search max value in dictionary and return key for this max
+# function search max value in dictionary and return keys for this max
 def find_max(dict):
-    max_key = [key for key, val in dict.items() if val == max(dict.values())]
+    max_value = max(dict.values())
+    max_key = [key for key, val in dict.items() if val == max_value]
     print(' '.join(map(str, max_key)))
     return max_key
 
 
 # function change list to dictionary.
 # For key set first element for value set list with next elements
 # eg. return
-# {'kot.j' : ['r', 'w',  'x'], 'pies.l' : ['r'], 'mysz' : ['r', 'x']}
+# {'kot.j' : ['r', 'w',  x'], 'pies.l' : ['r'], 'mysz' : ['r', 'x']}
 def change_to_dict(mylist):
     inventory = {}
     for i in range(len(mylist)):
         i_elem = mylist[i]
         split_elements = i_elem.split(' ')
-        key_elem = split_elements[0]
-        del split_elements[0]
-        values_elem = split_elements
-        inventory.setdefault(key_elem, values_elem)
+        key, *values = split_elements
+        # -------
+        # doesn't work, when put the same key
+        # inventory.setdefault(key_elem, values_elem)
+        # data wouldn't be save
+        # -------
+        # when put the same key, data are overwritten
+        inventory.update({key: values})
```

Comment:
Again, `inventory[key] = values` will do the same thing. `update()` is meant to be used like `old_version_of_the_dict.update(new_entries_which_may_overwrite_some_old_ones)`